.github/scripts/build_diff_payload.py

```
@@ -0,0 +1,132 @@
+def _read_at(head: str, path: str) -> str | None:
+    """Read a file's content at a specific commit, regardless of what's
+    currently checked out in the working tree.
+
+    Uses ``git show <head>:<path>``, which pulls the blob from the
+    object store. Reading from disk via ``pathlib`` would only work if
+    the runner had already checked out ``head``; this is more robust
+    and lets the script be exercised locally against historical
+    commits without checking them out first. Returns None if the path
+    doesn't exist at ``head`` (e.g. rare rename edge cases).
+    """
+    proc = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)